### DISCUSSION TITLE
Solicitação de Recorrência

### DISCUSSION THREAD

**Discussion by wfreitasdev** [bacen/pix-api#614] [Q&A]
Pelo que vi no swagger na versão 2.7.0 o PATCH do endpoint /solicrec/{idSolicRec} só tem uma propriedade no request que é o status, e o único valor possível é CANCELADA. Isto está certo?

**Comment by rodri-oliveira-dev** [comment]
Sim, pelo contrato esse comportamento é intencional.

Nesse endpoint o PATCH não está sendo usado como uma atualização genérica da SolicRec. Ele representa especificamente a operação de cancelamento da solicitação, por isso o único status aceito no request é `CANCELADA`.

Isso fica mais claro olhando a regra de negócio do próprio endpoint: uma SolicRec só pode ser cancelada quando estiver em um dos estados em que o cancelamento ainda faz sentido, como `CRIADA`, `ENVIADA` ou `RECEBIDA`.

As demais mudanças de estado da solicitação fazem parte do fluxo da recorrência e não são alterações arbitrárias que o recebedor pode fazer através desse PATCH.

Então eu trataria esse endpoint mais como uma transição de estado do domínio exposta via PATCH do que como um "partial update" convencional do recurso.
